cypress/integration/CTS/SearchResultsPageTests/index.js

```diff
@@ -0,0 +1,189 @@
+// <reference types="Cypress" />
+import { Given, And, Then, When } from 'cypress-cucumber-preprocessor/steps';
+
+Then('page title is {string}', (title) => {
+    cy.get('#NCI-CTS-root').find('h1').should('contain', title);
+});
+
+And('trial info displays {string}', (infoText) => {
+    cy.get('.all-trials').invoke('text').then((text) => {
+        const resultsText = text.split('\n').join(' ').split('\t').join('');
+        const regex = new RegExp(infoText);
+        expect(resultsText).to.match(regex);
+    });
+});
+
+And('{string} button is displayed', (expectedButton) => {
+    cy.get('.cts-results-header button').should('be.visible').and('contain.text', expectedButton);
+});
+
+
+When('user clicks on {string} button', (searchCriteriaButton) => {
+    cy.get('button.cts-accordion__button').contains(searchCriteriaButton).click();
+});
+
+And('search criteria table displays the following', (dataTable) => {
+    cy.get('tbody > tr').should('have.length', dataTable.hashes().length);
+    //index assures correct order
+    let index = 0;
+    for (const { Category, Selection } of dataTable.hashes()) {
+        cy.get('tbody tr th').eq(index).should('have.text', Category);
+        cy.get('tbody tr td').eq(index).should('have.text', Selection);
+        index++;
+    }
+});
+
+And('{string} link has a href {string}', (linkText, linkHref) => {
+    cy.get('a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+And('{string} pager is displayed', (pagerPosition) => {
+    cy.get(`div.results-page__control.--${pagerPosition} .results-page__pager`).should('be.visible');
+});
+
+And('result list is displayed', () => {
+    cy.get('div.results-list').should('be.visible');
+});
+
+And('each result item is a link', () => {
+    cy.document().then((doc) => {
+        const titleLocator = doc.querySelectorAll('.results-list-item__title');
+        for (let i = 0; i < titleLocator.length - 1; i++) {
+            cy.get(titleLocator[i]).should('be.visible');
+            cy.get(titleLocator[i]).find('a').should('have.attr', 'href');
+        }
+    });
+});
+
+And('there are {int} delighters present', (count) => {
+    cy.get('.cts-delighter-container div[class^="delighter"]:visible').should('have.length', count);
+});
+
+And('the following delighters are displayed', (dataTable) => {
+    for (const { delighter, href, title, text } of dataTable.hashes()) {
+        cy.get(`div[class="delighter ${delighter}"]:visible`).as('delighter');
+        cy.get('@delighter').find('a').first().should('have.attr', 'href', href);
+        cy.get('@delighter').find('h4').first().should('have.text', title);
+        cy.get('@delighter').find('p').first().should('have.text', text);
+    }
+});
+
+And('chat pop up is displayed', () => {
+    cy.wait(30000);
```

Comment:
So instead of explicitly waiting for 30 seconds, it is better to keep trying to find this element within 30 sec timeout . So it will retry until its found ( which is hopefully less than 30 sec)
e.g. 
 cy.get('#ProactiveLiveHelpForCTSPrompt', { timeout: 30000 }).should('be.visible');
 
 remove cy.wait

Reply:
Done